Controller/Adminhtml/Config/ResetConfig.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Wizzy\Search\Controller\Adminhtml\Config;
+
+use Magento\Framework\App\Config\Storage\WriterInterface;
+
+class ResetConfig extends \Magento\Framework\App\Action\Action
+{
+    protected $configWriter;
+    protected $cacheTypeList;
+    protected $messageManager;
+    protected $storeManager;
+    protected $request;
+
+    public function __construct(
+        \Magento\Framework\App\Request\Http $request,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Magento\Framework\Message\ManagerInterface $messageManager,
+        \Magento\Framework\App\Cache\TypeListInterface $cacheTypeList,
+        \Magento\Framework\App\Action\Context $context,
+        WriterInterface $configWriter
+    ) {
+
+        $this->storeManager = $storeManager;
+        $this->messageManager = $messageManager;
+        $this->cacheTypeList = $cacheTypeList;
+        $this->configWriter = $configWriter;
+        $this->request = $request;
+        return parent::__construct($context);
+    }
+
+    public function resetConfig(string $path)
+    {
+        $scopeId = $this->request->getParam('store');
+        $this->configWriter->save($path, $value = "", $scope = "stores", (int)$scopeId);
+    }
+
+    public function execute()
+    {
+        $this->resetConfig('wizzy_store_credentials/store_credentials/store_id');
```

Comment:
Take these paths from CONST variables we've defined in https://github.com/wizzy-ai/wizzy-search-magento-2/blob/dev/Services/Store/StoreCredentialsConfig.phpb